tests/baselines/reference/isolatedDeclarationErrorsObjects.errors.txt

```diff
@@ -0,0 +1,161 @@
+isolatedDeclarationErrorsObjects.ts(7,8): error TS9013: Expression type can't be inferred with --isolatedDeclarations.
+isolatedDeclarationErrorsObjects.ts(12,12): error TS9013: Expression type can't be inferred with --isolatedDeclarations.
+isolatedDeclarationErrorsObjects.ts(16,12): error TS9013: Expression type can't be inferred with --isolatedDeclarations.
+isolatedDeclarationErrorsObjects.ts(21,5): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations.
+isolatedDeclarationErrorsObjects.ts(24,5): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations.
+isolatedDeclarationErrorsObjects.ts(25,8): error TS9013: Expression type can't be inferred with --isolatedDeclarations.
+isolatedDeclarationErrorsObjects.ts(29,9): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations.
+isolatedDeclarationErrorsObjects.ts(32,9): error TS9008: Method must have an explicit return type annotation with --isolatedDeclarations.
+isolatedDeclarationErrorsObjects.ts(40,9): error TS7032: Property 'singleSetterBad' implicitly has type 'any', because its set accessor lacks a parameter type annotation.
+isolatedDeclarationErrorsObjects.ts(40,25): error TS7006: Parameter 'value' implicitly has an 'any' type.
+isolatedDeclarationErrorsObjects.ts(40,25): error TS9009: At least one accessor must have an explicit return type annotation with --isolatedDeclarations.
+isolatedDeclarationErrorsObjects.ts(64,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
+isolatedDeclarationErrorsObjects.ts(65,5): error TS9014: Computed properties must be number or string literals, variables or dotted expressions with --isolatedDeclarations.
+isolatedDeclarationErrorsObjects.ts(75,5): error TS9015: Objects that contain spread assignments can't be inferred with --isolatedDeclarations.
+isolatedDeclarationErrorsObjects.ts(77,5): error TS9016: Objects that contain shorthand properties can't be inferred with --isolatedDeclarations.
+isolatedDeclarationErrorsObjects.ts(84,9): error TS9015: Objects that contain spread assignments can't be inferred with --isolatedDeclarations.
+
+
+==== isolatedDeclarationErrorsObjects.ts (16 errors) ====
+    export let o = {
+        a: 1,
+        b: ""
+    }
+    
+    export let oBad = {
+        a: Math.random(),
+           ~~~~~~~~~~~~~
+!!! error TS9013: Expression type can't be inferred with --isolatedDeclarations.
+!!! related TS9027 isolatedDeclarationErrorsObjects.ts:6:12: Add a type annotation to the variable oBad.
+!!! related TS9035 isolatedDeclarationErrorsObjects.ts:7:8: Add a type assertion to this expression to make type type explicit.

```

Comment:
Suggesting a cast to fix this feels pretty unsafe. `satisfies` would be safer, but because of how that works, you can't pull its' type node, so isn't useful. Like, nothing is going to tell you off for writing `Math.random() as 0`, since casts aren't type annotations - they're just a place the emitter can read a type node from. I don't think we should suggest adding casts as the right fix for arbitrary expressions - annotations or bust, otherwise we're suggestion you make your code unchecked and unsafe.

Reply:
> Suggesting a cast to fix this feels pretty unsafe. `satisfies` would be safer, but because of how that works, you can't pull its' type node, so isn't useful. Like, nothing is going to tell you off for writing `Math.random() as 0`, since casts aren't type annotations - they're just a place the emitter can read a type node from. I don't think we should suggest adding casts as the right fix for arbitrary expressions - annotations or bust, otherwise we're suggestion you make your code unchecked and unsafe.

Can we suggest `satisfies T as T`?  From my experience having some sort of inline annotation is very useful for large object literals.

Reply:
> Can we suggest satisfies T as T? From my experience having some sort of inline annotation is very useful for large object literals.

That's safe enough that I'd be OK with it - it at least ensures you're not unsafely downcasting, since `Math.random() satisfies 0` errors, so if your expression type changes in some way, the cast will get an error if it becomes unsafe.

Reply:
Changed suggestion. @h-joo Could you please change the code fix to also do this.